### PR TITLE
tools: use fopen()/fread()/fwrite() instead of open()/read()/write() for file upload

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: libnodegl/utils.c [hunk 1]
```diff
@@ -153,3 +163,29 @@ void ngli_thread_set_name(const char *name)
...
+    if (file_handle == INVALID_HANDLE_VALUE)
+        return NGL_ERROR_IO;
+
+    LARGE_INTEGER file_size;
+    if (!GetFileSizeEx(file_handle, &file_size)) {
+        CloseHandle (file_handle);
+        return NGL_ERROR_IO;
+    }
+    *size = file_size.QuadPart;
+    CloseHandle (file_handle);
```
Comment (cboesch-gpsw) [inline libnodegl/utils.c:180]: don't forget to fix the spacing here too

File: ngl-tools/ngl-ipc.c [hunk 2]
```diff
@@ -87,13 +89,27 @@ static const struct opt options[] = {
...
 static int get_filesize(const char *filename, int64_t *size)
 {
+#ifdef _WIN32
+    HANDLE file_handle = CreateFile(TEXT(filename), GENERIC_READ, 0, NULL, OPEN_EXISTING, FILE_ATTRIBUTE_NORMAL, NULL);
+    if (file_handle == INVALID_HANDLE_VALUE)
+        return NGL_ERROR_IO;
+
+    LARGE_INTEGER file_size;
+    if (!GetFileSizeEx(file_handle, &file_size)) {
+        CloseHandle (file_handle);
```
[the remark this answers is not in the document]
Comment (mbouron-gpsw) [reply ngl-tools/ngl-ipc.c]: Fixed.

File: ngl-tools/ngl-ipc.c [hunk 3]
```diff
@@ -87,13 +89,27 @@ static const struct opt options[] = {
...
+        return NGL_ERROR_IO;
+    }
+    *size = file_size.QuadPart;
+    CloseHandle (file_handle);
```
[the remark this answers is not in the document]
Comment (mbouron-gpsw) [reply ngl-tools/ngl-ipc.c]: Fixed.

File: libnodegl/utils.c (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/utils.c]: Fixed.

File: ngl-tools/ngl-ipc.c [hunk 4]
```diff
@@ -28,12 +28,14 @@
...
+#include <Handleapi.h>
+#include <Fileapi.h>
 #else
 #include <sys/socket.h>
 #include <netdb.h>
-#endif
 #include <unistd.h>
 #include <sys/stat.h>
+#endif
 #include <fcntl.h>
```
[the remark this answers is not in the document]
Comment (mbouron-gpsw) [reply ngl-tools/ngl-ipc.c]: I've removed `fcntl.h` in a later commit (when we actually port open/* to fopen/*).

**Review by cboesch-gpsw — APPROVED** [review]

**Pull request merged**